# Teste Qui-Quadrado

In [2]:
from scipy.stats import chi2_contingency
import numpy as np
import math

In [3]:
tabela = np.array([[30, 20], [22, 28]])

In [4]:
tabela.shape # Formato de matriz

(2, 2)

In [6]:
_, p, _, _ = chi2_contingency(tabela)
chi2_contingency(tabela)

(1.9631410256410255,
 0.16117778944980238,
 1,
 array([[26., 24.],
        [26., 24.]]))

Logo, o p = 0.16117778944980238

 Em baixo, a frequencia esperada é dado pela tabela seguinte. 

In [7]:
alpha = 0.05

if p <= alpha:
    print("Hipotese nula rejeitada")
else:
    print("Hipotese alternativa rejeitada")

Hipotese alternativa rejeitada


Logo, o valor atual é o usado. Logo, não há diferença estatistica entre os dados.